ha/core/system_health/system_health.py

```diff
@@ -420,7 +420,7 @@ def process_event(self, healthevent: HealthEvent):
                     # Update the current health value itself.
                     latest_health = EntityHealth.read(current_health)
                     if stored_genration_id and (stored_genration_id != incoming_generation_id):
-                        if incoming_health_status == HEALTH_EVENTS.ONLINE.value:
+                        if incoming_health_status in [HEALTH_EVENTS.ONLINE.value, status]:
```

Comment:
This will always return true since incoming_health_status and status both are mapping to status in healthevent

Reply:
I think status maps to already stored status.

Reply:
Possibility of status here can be 'starting' or 'online'. So considering both will be fine.
I agree with this change.

Reply:
But, consider replicaset down case. 
1. When replicaset down, node status will be failed in system health.
2. Now, replicaset up, there is change in generation id and also incoming node status is online. 
So, this if will be passed and pod_restart count will be changed. But, in the replicaset scenario, we don't want that..!!

Reply:
once you did the required changes also write the comment for cases in which the restart count needs to be set/updated.
So this will make easy for someone to understand

Reply:
please review new change.

Reply:
I cannot resolve my comment alone if I resolve the whole conversation will get resolved,.
I am OK with changes let @ArchanaLimaye  resolve this conversion if she is OK as she is started this conversation.
CC @ArchanaLimaye